src/extension/exponent/exponentPlatform.ts

```diff
@@ -36,63 +37,67 @@ export class ExponentPlatform extends GeneralMobilePlatform {
-                (message, password) => {
-                    return Q.Promise((resolve, reject) => {
-                        vscode.window.showInputBox({ placeHolder: message, password: password })
-                            .then(login => {
-                                resolve(login || "");
-                            }, reject);
-                    });
-                },
-                (message) => {
-                    return Q.Promise((resolve, reject) => {
-                        const okButton =  { title: "Ok" };
-                        const cancelButton =  { title: "Cancel", isCloseAffordance: true };
-                        vscode.window.showInformationMessage(message, {modal: true}, okButton, cancelButton)
-                            .then(answer => {
-                                if (answer === cancelButton) {
-                                    reject(ErrorHelper.getInternalError(InternalErrorCode.UserCancelledExpoLogin));
-                                }
-                                resolve("");
-                            }, reject);
-                    });
-                }
-            )
-                .then(() =>
-                    XDL.setOptions(this.projectPath, { packagerPort: this.packager.port })
-                )
-                .then(() =>
-                    XDL.startExponentServer(this.projectPath)
-                )
-                .then(() =>
-                    XDL.startTunnels(this.projectPath)
-                )
-                .then(() =>
-                    XDL.getUrl(this.projectPath, { dev: true, minify: false })
-                ).then(exponentUrl => {
-                    return "exp://" + url.parse(exponentUrl).host;
-                })
-                .catch(reason => {
-                    return Q.reject<string>(reason);
-                })
-                .then(exponentUrl => {
-                    let exponentPage = vscode.window.createWebviewPanel("Expo QR Code", "Expo QR Code", vscode.ViewColumn.Two, { });
-                    exponentPage.webview.html = this.qrCodeContentProvider.provideTextDocumentContent(vscode.Uri.parse(exponentUrl));
-                    return exponentUrl;
-                })
-                .then(exponentUrl => {
-                    if (!exponentUrl) {
-                        return Q.reject<void>(ErrorHelper.getInternalError(InternalErrorCode.ExpectedExponentTunnelPath));
-                    }
-                    this.exponentTunnelPath = exponentUrl;
-                    const outputMessage = localize("ApplicationIsRunningOnExponentOpenToSeeIt", "Application is running on Expo. Open your Expo app at {0} to see it.", this.exponentTunnelPath);
-                    this.logger.info(outputMessage);
+        return ReactNativeProjectHelper.getReactNativeVersionFromProjectPackage(this.runOptions.projectRoot)
+            .then(version => {
+                TelemetryHelper.addReactNativeVersionToEventProperties(version, extProps);
```

Comment:
Should we add Expo version in such case? Also, what if we send both expo and react-native packages versions?

Reply:
I changed this behavior. Now we use React Native version from `runOptions` property.